`let...else` statements are unstable in  solana-frozen-abi-macro-1.16.0

#### Problem
<!--
  The GitHub issue tracker exists to track issues
  that affect the development of Solana itself.

  If you need technical support using Solana, building
  an app, or running a validator, don't open an issue here.

  Instead, post your question to the Solana Stack Exchange:
  https://solana.stackexchange.com/questions/ask
-->
While compiling anchor-lang = "0.28.0" project have got this error:
.cargo/registry/src/github.com-1ecc6299db9ec823/solana-frozen-abi-macro-1.16.0/src/lib.rs:422:5
   
422      let Some(expected_digest) = expected_digest else {
423           return Error::new_spanned(
424              TokenStream2::from(item),
425              "the required \"digest\" = ... attribute is missing.",
...
428          .into()
429      };
   
    = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
    = help: add `#![feature(let_else)]` to the crate attributes to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `solana-frozen-abi-macro` due to previous error
My rustc -V: rustc 1.70.0 (90c541806 2023-05-31)

This isn't an anchor specific issue. I can reproduce it attempting to build using rust 1.70 and dependency solana-program 1.16.1:

```
cargo new test-solana
cd test-solana
cargo add solana-program
cargo build-sbf
```
Compiling solana-frozen-abi-macro v1.16.1
error[E0658]: `let...else` statements are unstable
   --> /home/marshall/.cargo/registry/src/github.com-1ecc6299db9ec823/solana-frozen-abi-macro-1.16.1/src/lib.rs:422:5
    |
422 | /     let Some(expected_digest) = expected_digest else {
423 | |         return Error::new_spanned(
424 | |             TokenStream2::from(item),
425 | |             "the required \"digest\" = ... attribute is missing.",
...   |
428 | |         .into()
429 | |     };
    | |______^
    |
    = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
    = help: add `#![feature(let_else)]` to the crate attributes to enable

--- a/ci/test-stable.sh
+++ b/ci/test-stable.sh
@@ -58,6 +58,20 @@ test-stable-sbf)
   # SBF solana-sdk legacy compile test
   "$cargo_build_sbf" --manifest-path sdk/Cargo.toml
 
+  # Ensure the minimum supported "rust-version" matches platform tools to fail
+  # quickly if users try to build with an older platform tools install
+  cargo_toml=sdk/program/Cargo.toml
+  source "scripts/read-cargo-variable.sh"
+  crate_rust_version=$(readCargoVariable rust-version $cargo_toml)
+  platform_tools_rust_version=$("$cargo_build_sbf" --version | grep rustc)
+  platform_tools_rust_version=$(echo "$platform_tools_rust_version" | cut -d\  -f2) # Remove "rustc " prefix from a string like "rustc 1.68.0-dev"
+  platform_tools_rust_version=$(echo "$platform_tools_rust_version" | cut -d- -f1)  # Remove "-dev" suffix from a string like "1.68.0-dev"
+
+  if [[ $crate_rust_version != "$platform_tools_rust_version" ]]; then
+    echo "Error: Update 'rust-version' field in '$cargo_toml' from $crate_rust_version to $platform_tools_rust_version"
+    exit 1
+  fi
+
   # SBF C program system tests
   _ make -C programs/sbf/c tests
   if need_to_upload_test_result; then

--- a/sdk/program/Cargo.toml
+++ b/sdk/program/Cargo.toml
@@ -9,6 +9,7 @@ repository = { workspace = true }
 homepage = { workspace = true }
 license = { workspace = true }
 edition = { workspace = true }
+rust-version = "1.68.0" # solana platform-tools rust version
 
 [dependencies]
 bincode = { workspace = true }

